lib/mpl_toolkits/mplot3d/axes3d.py

```diff
@@ -1502,6 +1503,23 @@ def _calc_coord(self, xv, yv, renderer=None):
         p2 = p1 - scale*vec
         return p2, pane_idx
 
+    def _arcball(self, p):
+        """
+        Convert a point p = [x, y] to a point on a virtual trackball
+        This is Ken Shoemake's arcball
+        See: Ken Shoemake, "ARCBALL: A user interface for specifying
+        three-dimensional rotation using a mouse." in
+        Proceedings of Graphics Interface '92, 1992, pp. 151-156,
+        https://doi.org/10.20380/GI1992.18
+        """
+        p = 2 * p
+        r = p[0]**2 + p[1]**2
+        if r > 1:
+            p = np.concatenate(([0], p/math.sqrt(r)))
+        else:
+            p = np.concatenate(([math.sqrt(1-r)], p))
+        return p
```

Comment:
I find the compnent notation simpler to read. It's probably also a bit faster than small numpy arrays (but haven't checked). Please double check whether I've got the math right.
```suggestion
        x *= 2
        y *= 2
        r = x**2 + y**2
        if r > 1:
            return np.array([0, x/math.sqrt(r), y/math.sqrt(r)])
        else:
            return np.array([math.sqrt(1-r), x, y])
```
Not sure whether returning a tuple would be better than a numpy array, so I've left the numpy array.